manager/orchestrator/updater.go

```diff
@@ -49,7 +49,7 @@ func (u *UpdateSupervisor) Update(ctx context.Context, cluster *api.Cluster, ser
 	id := service.ID
 
 	if update, ok := u.updates[id]; ok {
-		if !update.isServiceDirty(service) {
+		if reflect.DeepEqual(service.Spec, update.newService.Spec) {
```

Comment:
Alternatively, if the better way to do things is the way it's done here, `isServiceDirty` is unused and should be removed.